### ISSUE TITLE
Virtual without getter returns `null`

### ISSUE THREAD

**Issue by JasonCust** [Automattic/mongoose#6223]
Issue began in mongoose version 4.11.5. Possibly introduced by the fix for #5473.

**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

If a virtual path only includes a setter (odd design, but that is neither here nor there 😉), as of v4.11.5 it will now return a `null` value for the virtual path. 

**If the current behavior is a bug, please provide the steps to reproduce.**

```javascript
const personSchema = new mongoose.Schema({
  name: { type: String },
  children: [{
    name: { type: String }
  }]
}, {
  toObject: { getters: true, virtuals: true },
  toJSON: { getters: true, virtuals: true }
});

personSchema.virtual('favoriteChild').set(function (v) {
  return this.set('children.0', v);
});

personSchema.virtual('heir').get(function () {
  return this.get('children.0');
});

const Person = mongoose.model('Person', personSchema);

const person = new Person({
  name: 'Anakin'
});

console.log(JSON.stringify(person, null, 2));
// Outputs:
{
  "name": "Anakin",
  "_id": "5aa18778ecad0fa061209c17",
  "children": [],
  "id": "5aa18778ecad0fa061209c17",
  "favoriteChild": null
}
```
**What is the expected behavior?**
```javascript
// Outputs:
{
  "name": "Anakin",
  "_id": "5aa18778ecad0fa061209c17",
  "children": [],
  "id": "5aa18778ecad0fa061209c17"
}
```

**Please mention your node.js, mongoose and MongoDB version.**

All mongoose versions since 4.11.5

**Comment by lineus** [comment]
Thanks for the awesome repro script @JasonCust !

I verified that this did change between 4.11.4 and 4.11.5 and persists on 5.0.9.
```
InspiredMacPro:mongoose4 lineus$ npm install mongoose@4.11.4 >/dev/null 2>&1
InspiredMacPro:mongoose4 lineus$ ./6223.js 
{
  "name": "Anakin",
  "_id": "5aa672e576d260d61c514490",
  "children": [],
  "id": "5aa672e576d260d61c514490"
}
^C
InspiredMacPro:mongoose4 lineus$ npm install mongoose@4.11.5 >/dev/null 2>&1
InspiredMacPro:mongoose4 lineus$ ./6223.js 
{
  "name": "Anakin",
  "_id": "5aa672fe8a6d9dd637a2fc7d",
  "children": [],
  "id": "5aa672fe8a6d9dd637a2fc7d",
  "favoriteChild": null
}
^C
InspiredMacPro:mongoose4 lineus$ cd ../mongoose5
InspiredMacPro:mongoose5 lineus$ ./6223.js 
{
  "name": "Anakin",
  "children": [],
  "_id": "5aa673384e1aced658075721",
  "favoriteChild": null,
  "id": "5aa673384e1aced658075721"
}
^C
```
From looking at the commits, the relevant commit is probably: 2ede28b1fe4db0895a4dcb19275adc8349dc8e24

**Comment by vkarpov15** [comment]
Fixed in c6295cb, will be in 4.13.12
